src/holofuel/pages/Inbox/Inbox.js

```diff
@@ -35,10 +36,41 @@ function useDecline () {
   })
 }
 
+function useFetchCounterparties () {
+  const { data: { holofuelActionableTransactions = [] } = {} } = useQuery(HolofuelActionableTransactionsQuery)
+
+  const { loading, error, data: { holofuelInboxCounterparties } = {}, client } = useQuery(HolofuelInboxCounterpartiesQuery)
+
+  if (holofuelInboxCounterparties) {
+    const filterTransactionsByAgentId = (agent, txListType) => txListType.filter(transaction => transaction.counterparty.id === agent.id)
+    const updateTxListCounterparties = (txListType, counterpartyList) => counterpartyList.map(agent => {
+      const matchingTx = filterTransactionsByAgentId(agent, txListType)
+      return matchingTx.map(transaction => { Object.assign(transaction.counterparty, agent); return transaction })
+    })
+
+    const result = _.flatten(updateTxListCounterparties(holofuelActionableTransactions, holofuelInboxCounterparties))
+
+    client.writeQuery({
+      query: HolofuelActionableTransactionsQuery,
+      data: {
+        holofuelActionableTransactions: { ...result }
+      }
+    })
+  }
+
+  let response
+  if (loading) response = { loading: true }
+  if (error) response = { error: `Error: ${error}` }
+  else response = holofuelInboxCounterparties
+  return response
+}
+
```

Comment:
This is an interesting approach, and still not the one I meant by "using the cache". I think the easiest thing is for us to pair on this for 20 minutes. I'm not sure my approach will end up being better but I'd still like to see it implemented so we can compare.

Reply:
Happy to share ideas and compare options. :100: